### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by jamiebuilds**
Was bored and found this repo, I cleaned it up with some more ES6 and use bluebird/promises more effectively.

I removed the `Publish` class because it wasn't really accomplishing anything other than being an awkward state container.

Since there aren't any tests I haven't really been able to test this code, but the changes I made should hopefully make it easier to test.

### CODE REVIEW

**Comment by jenslind** [comment]
:star2: super nice, thanks! One thing though, I'm using `standard` ;) But I can fix the warnings when merging. Just gonna test it first.

File: src/index.js [hunk 1]
```diff
@@ -1,113 +1,79 @@
...
-    if (!Array.isArray(self.opts.output)) self.opts.output = self.opts.output.replace(/ /g, '').split(',')
+function getRepo(pkg) {
+  let url = pkg.repository.url.split('/')
+  return url[3] + '/' + url[4].replace(/\.[^/.]+$/, '')
+}
 
-    return new Promise(function (resolve, reject) {
-      if (self.opts.app.length !== self.opts.output.length) reject(new Error('Output length does not match app length'))
+function getTag() {
+  return `v${version}`
```
Comment (jenslind) [inline src/index.js]: `version` is undefined

File: src/index.js [hunk 2]
```diff
@@ -1,113 +1,79 @@
...
 
-  // Get repo from package.json
-  _getRepo () {
-    let pkg = this._loadPackageJson()
-
-    let url = pkg.repository.url.split('/')
-    return url[3] + '/' + url[4].replace(/\.[^/.]+$/, '')
-  }
+  return Promise.resolve(app).map((item, index) => {
+    let outputZip = (path.extname(output[i]) === '.zip') ? output[i] : output[i] + '.zip'
```
Comment (jenslind) [inline src/index.js]: `i` => `index`

File: src/index.js [hunk 3]
```diff
@@ -1,113 +1,79 @@
...
+      throw new Error('Unable to compress app.')
+    })
+  })
+}
 
-    let version = pkg.version
-    return 'v' + version
-  }
+export function release({ token, repo, tag, name, output }) {
+  return publishReleaseAsync({
```
Comment (jenslind) [inline src/index.js:76]: This does not work. It creates the release but it seems it fails to upload the assets. Maybe have something todo with bluebirds `promisify`?

If I pass `--app=myApp.app` i get:

```
fs.js:849
  return binding.stat(pathModule._makeLong(path));
                 ^
Error: ENOENT: no such file or directory, stat 't'
    at Error (native)
    at Object.fs.statSync (fs.js:849:18)
    at /Users/jenslind/Nodes/electron-release/node_modules/publish-release/index.js:80:23
...
```

https://github.com/remixz/publish-release/blob/master/index.js#L80

If I use `--app=./myApp.app` I get:

```
Error: EISDIR: illegal operation on a directory, read
```

Comment (jamiebuilds) [inline src/index.js:76]: Can you paste the full stack trace?

Comment (jenslind) [inline src/index.js:76]: http://pastebin.com/1qjBQ0mD

Comment (jamiebuilds) [inline src/index.js:76]: I was able to get it working over here: https://github.com/thejameskyle/__electron-release-test

I need to clean something up though.

**Comment by jamiebuilds** [comment]
This should be good to go

**Comment by jenslind** [comment]
:tada: Added you as an owner here and on npm.

**Pull request closed without merging**